### PR TITLE
use uint32_t for index

### PR DESCRIPTION
**Pull request by minggo**

### CODE REVIEW

File: native/cocos/core/assets/EffectAsset.h [hunk 1]
```diff
@@ -404,7 +404,7 @@ struct IBlockInfo {
 };
 
 struct ISamplerTextureInfo {
-    int32_t binding{-1};
+    uint32_t binding{0};
```
Comment (star-e) [inline native/cocos/core/assets/EffectAsset.h]: binding has to be 0xFFFFFFFF, which is the default (null) value.

File: native/cocos/core/assets/Material.h [hunk 2]
```diff
@@ -311,7 +311,7 @@ class Material : public Asset {
 protected:
     void update(bool keepProps = true);
     bool uploadProperty(scene::Pass *pass, const ccstd::string &name, const MaterialPropertyVariant &val);
-    void bindTexture(scene::Pass *pass, uint32_t handle, const MaterialProperty &val, index_t index = CC_INVALID_INDEX);
+    void bindTexture(scene::Pass *pass, uint32_t handle, const MaterialProperty &val, uint32_t index = 0);
```
Comment (star-e) [inline native/cocos/core/assets/Material.h:314]: ~~`uint32_t index` should be CC_INVALID_INDEX, which is 0xFFFFFFFF~~
We can use 0 as default value

File: native/cocos/core/assets/EffectAsset.h (hunk 1 above)
Comment (minggo) [reply native/cocos/core/assets/EffectAsset.h]: Yep, seems reasonable.

Comment (minggo) [reply native/cocos/core/assets/EffectAsset.h]: fixed

**Review by dumganhar — APPROVED** [review]

**Review by star-e — APPROVED** [review]

**Pull request merged**